client/blocks/inline-help/admin-sections.js

```diff
@@ -425,16 +433,20 @@ export function filterListBySearchTerm( searchTerm = '', collection = [], limit
 		'gi'
 	);
 
-	return collection
-		.filter( ( item ) => {
-			if ( searchRegex.test( item.title ) ) {
-				return true;
-			}
-			// Until we get the synonyms translated, just check when the language is `'en'`
-			return 'en' === getLocaleSlug()
-				? intersection( item.synonyms, searchTermWords ).length > 0
-				: false;
-		} )
+	const exactMatches = collection.filter(
+		( item ) => item.title.toLowerCase() === searchTerm.toLowerCase()
+	);
+	const partialMatches = collection
+		.filter( ( item ) => ! exactMatches.includes( item ) )
+		.filter( ( item ) => ( searchRegex.test( item.title ) ? true : false ) );
+
+	const synonymMatches = collection
+		.filter( ( item ) => ! exactMatches.includes( item ) )
+		.filter( ( item ) => ! partialMatches.includes( item ) )
+		.filter( ( item ) =>
+			'en' === getLocaleSlug() ? intersection( item.synonyms, searchTermWords ).length > 0 : false
+		);
+	return [ ...exactMatches, ...partialMatches, ...synonymMatches ]
```

Comment:
Love that feedback! I was actually carried away with the functional implementation! :D I will fix this.

Reply:
Done.

Reply:
Looks much better. Thanks for taking the advice :)

Side note: For a functional language implementing lazy evaluation, it often makes sense to write code that appears to go over the same data structure for multiple times, because the lazy evaulation would bind the computation into a list of "thunks", and only evaluate each thunk when necessary. I'd argue that this fundamental difference of runtime system is exactly why many FP paradigms don't make sense in imperative languages like JS. :)